src/frontend/src/handler/create_source.rs

```diff
@@ -110,45 +110,35 @@ async fn extract_avro_table_schema(
         .collect_vec())
 }
 
-/// Map an Avro schema to a relational schema. And extract primary key columns.
-async fn extract_upsert_avro_table_schema(
+/// Extract Avro primary key columns.
+async fn extract_upsert_avro_table_pk_columns(
     info: &StreamSourceInfo,
     with_properties: &HashMap<String, String>,
-) -> Result<(Vec<ColumnCatalog>, Vec<String>)> {
+) -> Result<Option<Vec<String>>> {
```

Comment:
give an empty vec instead?

Reply:
Sure I can do this, but it might need further refactoring that is still in discussion.